src/zip.ts

```diff
@@ -0,0 +1,119 @@
+import fs from 'fs';
+import path from 'path';
+import { execSync } from 'child_process';
+import { platform } from 'os';
+import { errCode } from './error';
+
+function validate(workspace: string, source: string, dest: string) {
+  if (!source.trim()) {
+    throw new Error('The source path cannot be empty.');
+  }
+  if (!dest.trim()) {
+    throw new Error('The destination file cannot be empty.');
+  }
+
+  // Verify the source folder exists and is not a file.
+  try {
+    const stats = fs.statSync(source);
+    if (!stats.isDirectory()) {
+      throw new Error('Invalid source folder: the source must be a directory.');
+    }
+  } catch (err) {
+    if (errCode(err) === 'ENOENT') {
+      throw new Error('Invalid source folder: not exist.');
+    }
+    throw new Error(`Failed to access source folder: ${err}`);
+  }
+
+  if (path.isAbsolute(source)) {
+    throw new Error('Invalid source folder: absolute path is not supported.');
+  }
+  if (isFolderOutside(source, workspace)) {
+    throw new Error(
+      'Invalid source folder: the source path is outside the user workspace.',
+    );
+  }
+  if (!dest.endsWith('.zip')) {
+    throw new Error('Invalid zip filename: Only .zip file is permitted.');
+  }
+}
+
+/**
+ * Checks if a folder is outside of a specified folder.
+ *
+ * @param {string} targetFolder The path to the target folder.
+ * @param {string} specifiedFolder The path to the specified folder.
+ * @returns {boolean} Returns true if the target folder is outside of the specified folder, false otherwise.
+ */
+export function isFolderOutside(
+  targetFolder: string,
+  specifiedFolder: string,
+): boolean {
+  // Resolve absolute paths.
+  const absoluteTargetFolder = path.resolve(targetFolder);
+  const absoluteSpecifiedFolder = path.resolve(specifiedFolder);
+
+  // Check if the target folder is the same as the specified folder (not outside).
+  if (absoluteTargetFolder === absoluteSpecifiedFolder) {
+    return false;
+  }
```

Comment:
True, but does that mean we'd allow our user to _potentially_ archive the entire workspace? Or is something else preventing that somewhere?

Reply:
hmm... make sense. Let me remove it to prevent that user behavior.